mmeval/metrics/coco_pose.py

```diff
@@ -0,0 +1,641 @@
+                    - 'keypoints' (np.ndarray): Shape (N, K, 2).
+                        The predicted keypoints coordinates of the instance.
+                        N: number of instances, K: number of keypoint.
+                        For topdown-style output, N is usually 1, while
+                        for bottomup-style output, N is the number of instances
+                        in the image.
+                    - 'keypoint_scores' (np.ndarray): Shape (N, K).
+                        The scores for all keypoints of all instances.
+                    - 'bbox_scores' (np.ndarray): Shape (N, ).
+                        The predicted scores of the bounding boxes.
+
+            groundtruths (Sequence[dict]): A sequence of dict. If load from
+                `ann_file`, the dict inside can be empty. Else, each dict
+                represents a groundtruths for an instance, with the following
+                keys:
+                    - 'img_id' (int): The image_id of the instance
+                    - 'width' (int): The width of the image.
+                    - 'height' (int): The height of the image.
+                    - 'crowd_index' (float, optional): Measure the crowding
+                        level of an image, defined in CrowdPose dataset
+                    - `raw_ann_info` (dict): The raw annotation information
+                        of the instances. It is worth mentioning that, to
+                        compute `COCOKeyPointDetection`, there are some
+                        required keys in the `raw_ann_info`:
+                            - `id`: the id to distinguish different annotations
+                            - `image_id`: the image id of this annotation
+                            - `category_id`: the category of the instance.
+                            - `bbox`: the bounding box of the instance
+                            - `keypoints`: the keypoints cooridinates with
+                                their visibilities. It need to be aligned
+                                with the official COCO format, e.g., a list
+                                with length N * 3, in which N is the number of
+                                keypoints. And each triplet represent the
+                                [x, y, visible] of the keypoint.
+                            - `iscrowd`: indicating whether the annotation is
+                                a crowd. It is useful when matching the
+                                detection results to the ground truth.
+                        There are some optional keys as well:
+                            - `area`: it is necessary when `self.use_area` is
+                                `True`
+                            - `num_keypoints`: it is necessary when
+                                `self.iou_type` is set as `keypoints_crowd`.
+        """
+        for prediction, groundtruth in zip(predictions, groundtruths):
```

Comment:
<img width="502" alt="image" src="https://user-images.githubusercontent.com/46705745/220401086-fdd2e85f-bbb5-498e-a9bb-d7d92e794a77.png">
I think the developer assumed both to be of instance-level. Also can be inferred from the unit test.
<img width="494" alt="image" src="https://user-images.githubusercontent.com/46705745/220400922-830194f5-27d8-4f87-9176-bcd89b4efda4.png">

Reply:
What's the meaning of `id` in `predictions`? Is there only 1 instance in each image, or there might be multiple instances in an image? How to match your detections with groundtruths without knowing the groundtruths, i.e. same number of detections as groundtruths?

I'm not familiar with the format, but seems like `raw_ann_info` in `groundtruth` covers all instances' labels in a single image. Please clarify if I misunderstood.

Reply:
There might be multiple instances in an image. For example, in `coco_pose_sample.json`, there are three entries with `"image_id": 40083`, because in the [image](http://images.cocodataset.org/val2017/000000040083.jpg), there are three people.

In `compute_metric`, groundtruths (a list of groundtruth instances) are represented by a COCO format json file, while predicted instances of the same image are grouped together and dumped into another json file. The evaluation results are calculated by `COCOeval` with the above two files. 

Each `ann` (`raw_ann_info`) is an instance, with information of the groundtruth keypoints and bboxes, etc.